synchro/synchrotest.py

```diff
@@ -334,17 +334,8 @@ def want_method(method, classname):
         "pymongo.mongo_client",
         "pymongo.database",
         "pymongo.srv_resolver",
-        "pymongo.synchronous.collection",
-        "pymongo.synchronous.client_session",
-        "pymongo.synchronous.command_cursor",
-        "pymongo.synchronous.change_stream",
-        "pymongo.synchronous.cursor",
-        "pymongo.synchronous.encryption",
-        "pymongo.synchronous.mongo_client",
-        "pymongo.synchronous.database",
```

Comment:
Could you explain these changes?

Reply:
Ah this is exactly the case I was worried about here: https://github.com/mongodb/mongo-python-driver/pull/1820#discussion_r1739317264

Reply:
A very accurate prediction, unfortunately I can't find a solution for this issue in Motor that doesn't rely on also changing PyMongo code.

Reply:
an option is to get the class name of the full mro and check for the desired name

Reply:
Something like `type(client).__name__ == "MongoClient"`?

Reply:
But also for subclasses:

```
>>> class Foo(MongoClient):
...    pass
...
>>> Foo.__mro__
(<class '__main__.Foo'>, <class 'pymongo.synchronous.mongo_client.MongoClient'>, <class 'pymongo.common.BaseObject'>, <class 'typing.Generic'>, <class 'object'>)
```

Reply:
We'd still run into the same issue with that, right? Since the import paths differ.